packages/app-aco/src/components/AdvancedSearch/QueryBuilder/adapters/QueryBuilderViewModel.ts

```diff
@@ -0,0 +1,14 @@
+import { makeAutoObservable } from "mobx";
+import { FieldDTO, QueryObjectDTO } from "../domain";
+
+export class QueryBuilderViewModel {
+    public queryObject: QueryObjectDTO;
+    public fields: FieldDTO[];
+    public invalidFields: Record<string, { isValid: boolean; message: string }> = {};
+
+    constructor(queryObject: QueryObjectDTO, fields: FieldDTO[]) {
+        this.queryObject = queryObject;
+        this.fields = fields;
+        makeAutoObservable(this);
+    }
+}
```

Comment:
This class is now redundant. You can simply keep the `queryObject` and `fields` in the Presenter, and assign them to the viewModel when/if necessary. This class does nothing, so you can get rid of it.